### PR TITLE
enable no-console eslint rule for project

### PR DESCRIPTION
**Pull request by jsnellbaker**
## Type of change
- [x] Code style update (formatting, local variables)
- [x] Other

## Description of change
There have been a number of PRs in the past where the unit tests included some rogue `console.log` statements and they were accidentally missed by the reviewers.

While not a big deal, it does end up polluting the console messages when we run the tests (especially the browserstack tests, since they get displayed for each browser), and it becomes somewhat tedious to track them down later and create another PR just to remove them.

I made this PR to enable the no-console rule for the eslint rules in our `test` directory to hopefully help us better catch these instances during the review process.  

I also updated the various files in the `test` directory using `console.log`; either removing them or disabling the rule.

**Update**
Per feedback, I have extended this lint logic to the project rather than just the tests directory.

### CODE REVIEW

File: test/spec/modules/onetagBidAdapter_spec.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ describe('onetag', function () {
         }
       });
     } catch (e) {
-      console.log('Error while parsing');
+      console.log('onetagBidAdapter tests: Error while parsing');  // eslint-disable-line
```
Comment (mkendall07) [inline test/spec/modules/onetagBidAdapter_spec.js]: any reason to keep these these?

Comment (jsnellbaker) [reply test/spec/modules/onetagBidAdapter_spec.js]: I thought at first they would be good to keep for the adapter's tests, but in retrospect they're not too helpful.  I pushed an update to remove them.

**Comment by jsnellbaker** [comment]
Hi @snapwich 

Expanding the scope does make sense, I'll try to make the necessary updates soon.

**Comment by jsnellbaker** [comment]
I pushed in the requested updates.  Please take another look when you get the chance.

Thanks!

**Comment by jsnellbaker** [comment]
Hey @snapwich @mkendall07 , can you please take another look and approve (if it looks good now)?  Thanks!

**Pull request merged**